test/integration/features/profile.feature

```diff
@@ -2,104 +2,251 @@
-     Given Minishift has state "Does Not Exist"
-      When executing "minishift start" succeeds
-      Then Minishift should have state "Running"
+  Scenario Outline: As user, I can't create profile with blank profile name
+    Given Profile "minishift" has state "Does Not Exist"
+     When executing "minishift profile set <profilename>" fails
+     Then stderr should contain
+      """
+      A profile name must be provided. Run `minishift profile list` for a list of existing profiles.
+      """
 
-Scenario: User should be able to list default profile 'minishift'
-   Given Minishift has state "Running"
-    When executing "minishift profile list" succeeds
-    Then stdout should contain
-    """
-    - minishift	Running		(Active)
-    """
-  Scenario: Getting default profile internal docker registry address
-     Given Minishift has state "Running"
-      When executing "minishift openshift registry" succeeds
-      Then stdout should be valid IP with port number
+  Examples: Empty profile name
+    | profilename |
+    | ''          |
+    | ' '         |
+    
+  Scenario Outline: As user, I can't create profile with special character in profile name
+    Given Profile "minishift" has state "Does Not Exist"
+     When executing "minishift profile set <profilename>" fails
+     Then stderr should contain
+      """
+      Profile names must consist of alphanumeric characters only.
+      """
+  
+  Examples: Bad profile name
+    | profilename |
+    | '.'         |
+    | '#$'        |
+    | ' test'     |
+    | '@test'     |
+    | '!profile'  |
+    | 'demo@1'    |
+    | '?aaa#'     |
+    | '%pro%'     |
+    | '&foo'      |
+    | '*te$st^'   |
+    | 'foo 123'   |
+    | 'foo    '   |
 
-Scenario: Starting Minishift with profile foo
-    When executing "minishift start --profile foo" succeeds
-    Then Minishift should have state "Running"
+  Scenario Outline: As user, I can create profile with alphanumeric character including '_' and '-' in profile name
+    NOTE: Though single character like a-z, A-Z or single digit between 0-9 are consider as alphanumeric, will not be supported here.
+    See https://github.com/minishift/minishift/issues/1748 for more details.
+    Given Profile "minishift" has state "Does Not Exist"
+     When executing "minishift profile set <profilename>" succeeds
+     Then stdout should contain
+      """
+      Profile <profilename> set as active profile.
+      """
+  
+  Examples: Good profile name
+    | profilename |
+    | '123foo'    |
+    | 'test-123'  |
+    | 'profile_'  |
+    | 'foo-'      |
+    | 'vm_45'     |
+    | '20'        |
+    | 'random001z'|
+    | '00_XYZ-'   |
+    | 'foo'       |
 
-  Scenario: Getting profile foo internal docker registry address
-     Given Minishift has state "Running"
-      When executing "minishift openshift registry" succeeds
-      Then stdout should be valid IP with port number
+  Scenario: As user, I can list all profile
```

Comment:
This should be part of the previous scenario since it really depends on it.

Reply:
Awesome..Scenario would be like
Scenario Outline: As user, I can create profile with alphanumeric character including '_' and '-' in profile name
[...]
And executing "minishift profile list" succeeds
Then stdout should match "-\s*<profilename>\s*Does Not Exist\s*\(Active\)"

Will check the implimentation